### ISSUE TITLE
A few quick questions!

### ISSUE THREAD

**Issue by mason0958** [cazala/synaptic#233]
1) What is the function of the learning rate when training a network/layers?
2) When building a neural network, how do you determine the minimum width of neurons and/or layers needed for the correct output?
3) Do I need to train the network the correct output for every possible input? i.e. I built a network earlier today and was training it to output 1 only when a specific input was given, however, it was giving the same output no matter what input I gave it. Do I need to train the network to output 0 for every other possible input?

I've been having a blast messing around with this library. Good work to everyone who's worked on it and I'd love to contribute to it in the future once I've mastered NN's.

THANKS!!

**Comment by wagenaartje** [comment]
1. The default function is _fixed_, so the learning rate stays constant over time. You can also pass on an array, with different learning rates, which it will divide equally over the iterations. 
2. You really can't. This is something that has to be done by trial and error. More info [here](https://stats.stackexchange.com/questions/181/how-to-choose-the-number-of-hidden-layers-and-nodes-in-a-feedforward-neural-netw)
3. No you don't. But in your case, you only have 1 test samples and there is no relationship between different test samples. Normally, if you train a neural network you provide _a lot_ of data, and train the network on 80% percent on it. Then you test how well the network performs on the other 20%, which it was never trained on. However, for small datasets, there is no guarantee that the network will also work for values that are not in the input samples.

**Comment by mason0958** [comment]
1) Ahh, ok, so that was bad wording on my part in regard to the first question. What I really meant was, what is the purpose of the learning rate/ what is the learning rate?

2) Gotchya, that's what I've been reading elsewhere and I've found some articles detailing general best practices to initially try so you don't get way off from the ideal architecture when testing to find it.

3) Yes, I see now. I never took into account the fact that ultimately NN's are used to recognize similar patterns within different datasets. In my case, I might as well have used a simple if statement lol.

That must have been a record response time for a GitHub repo issue; I'm glad to see you guys care about this project. 

Thanks again! This was incredibly helpful.

**Comment by wagenaartje** [comment]
The learning rate is a variable used during backpropagation, specificely during the weight and bias update. Basically, the algorithm calculates if a weight should be larger or smaller, basically a direction to go to for the weight (up/down). The learning rate tells the algorithm: we should move this much higher, not too much, so we don't surpass the target, and not too little, otherwise we won't learn anything.

> Training parameter that controls the size of weight and bias changes in learning of the training algorithm. [source](https://en.wikibooks.org/wiki/Artificial_Neural_Networks/Neural_Network_Basics#Learning_Rate)

More info [here](https://datascience.stackexchange.com/questions/410/choosing-a-learning-rate).

No problem :)
